scripts/linker/analyze_classification_links.py

```diff
@@ -0,0 +1,181 @@
+"""
+This script creates a summary and graphs derived from the classification results.
+
+It needs to be executed inside the directory which contains the classification links files
+"""
+
+from glob import glob
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+
+sns.set(rc={"lines.linewidth": 0.5})
+
+df = pd.DataFrame(columns=[
+    "Catalog", "Entity", "Model", "Count", "Mean", "STD",
+    "Min", "25%", "50%", "75%", "Max"
+])
+
+files = glob("*_links.csv.gz")
+all_links = pd.DataFrame(columns=[
+    "Catalog",
+    "Entity",
+    "Model",
+    "QID",
+    "TID",
+    "Prediction"
+])
+
+non_binary_classifiers = [
+    "voting_classifier_soft",
+    "logistic_regression",
+    "multi_layer_perceptron",
+    "naive_bayes",
+    "single_layer_perceptron",
+    "random_forest"
+]
+
+# For every "links" file
+for f in files:
+    catalog = f[:f.index("_")]
+
+    et = f[f.index("_") + 1:]
+    entity = et[:et.index("_")]
+
+    model = "_".join(et.split("_")[1:-1])
+
+    current_preds = pd.read_csv(f,
+                                header=None,
+                                names=['QID', 'TID', 'Prediction'],
+                                index_col=['QID', 'TID'])
+
+    current_preds["Catalog"] = catalog
+    current_preds["Entity"] = entity
+    current_preds["Model"] = model
+
+    all_links = all_links.append(current_preds.reset_index(), sort=False)
+
+    classfs = (current_preds['Prediction']
+               .describe()
+               .to_dict())
+
+    for k in classfs.keys():
+        classfs[k] = "%.6f" % classfs[k]
+
+    classfs['count'] = classfs['count'][:classfs['count'].index(".")]
+
+    # add the files statistics to the dataframe
+    df = df.append({
+        "Catalog": catalog,
+        "Entity": entity,
+        "Model": model,
+        "Count": classfs['count'],
+        "Mean": classfs['mean'],
+        "STD": classfs['std'],
+        "Min": classfs['min'],
+        "25%": classfs['25%'],
+        "50%": classfs['50%'],
+        "75%": classfs['75%'],
+        "Max": classfs['max']
+    }, ignore_index=True)
+
+# print all files' statistics
+df = df.sort_values(by=['Model', 'Catalog'])
+print(df.to_csv(index=False))
+
+###############
+# Get summaries of the statistics and print them
+summaries = []
+for _, gg in df.groupby('Model'):
```

Comment:
Yes you're right. But we would still need to process the result of `describe` to get the information we're interested in presenting (since `describe` yields much more information than what we're interested in). 

If you think it would be cleaner to use `df.describe()` anyway then let me know and I can change it.

Reply:
fair enough